File: src/main/scala/com/countrygamer/cgo/common/lib/util/Player.java
package com.countrygamer.cgo.common.lib.util;

import net.minecraft.block.Block;
import net.minecraft.client.Minecraft;
import net.minecraft.client.multiplayer.WorldClient;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.entity.player.EntityPlayerMP;
import net.minecraft.server.MinecraftServer;
import net.minecraft.util.ChatComponentText;
import net.minecraft.util.MathHelper;
import net.minecraft.world.World;

import java.util.List;
import java.util.UUID;

public class Player {

	public static void sendMessageToPlayer(EntityPlayer player, String message) {
		//if (player.worldObj.isRemote)
		player.addChatComponentMessage(new ChatComponentText(message));
	}

	public static boolean breakBlockAsPlayer(World world, EntityPlayer player, int x, int y, int z,
			Block block) {
		if (player == null || world == null)
			return false;
		WorldClient worldclient = Minecraft.getMinecraft().theWorld;
		worldclient.playAuxSFX(2001, x, y, z,
				Block.getIdFromBlock(block) + (worldclient.getBlockMetadata(x, y, z) << 12));

		int meta = world.getBlockMetadata(x, y, z);
		world.setBlockToAir(x, y, z);
		block.onBlockDestroyedByPlayer(world, x, y, z, meta);

		return true;
	}

	public static int getDirection(EntityPlayer player) {
		return MathHelper.floor_double((double) (player.rotationYaw * 4.0F / 360.0F) + 0.5D) & 3;
	}

	@SuppressWarnings("rawtypes")
	public static EntityPlayerMP getPlayerByUUID(UUID id) {
		List players = MinecraftServer.getServer().getConfigurationManager().playerEntityList;
		for (Object player1 : players) {
			if (player1 instanceof EntityPlayerMP) {
				EntityPlayerMP player = (EntityPlayerMP) player1;
				if (player.getUniqueID().equals(id)) {
					return player;
				}
			}
		}
		return null;
	}

	public static String getUsername(EntityPlayer player) {
		return player.getCommandSenderName(); // Also player.getGameProfile().getName()
	}

	public static EntityPlayerMP getPlayerByUsername(String name) {
		List players = MinecraftServer.getServer().getConfigurationManager().playerEntityList;
		for (Object player1 : players) {
			if (player1 instanceof EntityPlayerMP) {
				EntityPlayerMP player = (EntityPlayerMP) player1;
				if (player.getCommandSenderName().equals(name)) {
					return player;
				}
			}
		}
		return null;
	}

}

File: src/main/scala/com/temportalist/origin/library/common/utility/Player.scala
package com.temportalist.origin.library.common.utility

import net.minecraft.client.Minecraft
import net.minecraft.entity.player.{EntityPlayer, EntityPlayerMP}
import net.minecraft.server.MinecraftServer
import net.minecraft.util.ChatComponentText
import net.minecraftforge.fml.relauncher.{Side, SideOnly}

/**
 *
 *
 * @author TheTemportalist
 */
object Player {

	def getPlayer(senderNameORuuid: AnyRef): EntityPlayerMP = {
		val players: java.util.List[_] =
			MinecraftServer.getServer.getConfigurationManager.playerEntityList
		for (i <- 0 until players.size()) {
			players.get(i) match {
				case player: EntityPlayerMP =>
					if (player.getName.equals(senderNameORuuid) ||
							player.getUniqueID.equals(senderNameORuuid)) {
						return player
					}
				case _ =>
			}
		}
		null
	}

	def forceMoveInDirection(player: EntityPlayerMP, distance: Double,
			considerObstacles: Boolean): Unit = {
		val rotH: Double = Math.toRadians(player.rotationYaw)
		val rotV: Double = Math.toRadians(player.rotationPitch)
		val dx: Double = -Math.sin(rotH) * distance
		val dy: Double = Math.sin(rotV) * distance
		val dz: Double = Math.cos(rotH) * distance

		if (considerObstacles) {
			// todo how to determine dy correctly?
			player.moveEntity(dx, 0, dz)
		}
		else {
			Teleport.toPoint(player, player.posX + dx, player.posY + dy, player.posZ + dz)
		}
	}

	def message(player: EntityPlayer, message: String): Unit = {
		player.addChatComponentMessage(new ChatComponentText(message))
	}

	def getReachDistance(player: EntityPlayer): Double = {
		if (player.getEntityWorld.isRemote)
			this.getReach_client()
		else player match {
			case mp: EntityPlayerMP => this.getReach_server(mp)
			case _ => 5.0D
		}
	}

	@SideOnly(Side.CLIENT)
	private def getReach_client(): Double = Minecraft.getMinecraft.playerController.getBlockReachDistance

	private def getReach_server(mp: EntityPlayerMP): Double = mp.theItemInWorldManager.getBlockReachDistance

}
